protocols/floodsub/src/protocol.rs

```diff
@@ -93,21 +106,24 @@ where
 
 /// Reach attempt interrupt errors.
 #[derive(thiserror::Error, Debug)]
-pub enum FloodsubDecodeError {
-    /// Error when reading the packet from the socket.
-    #[error("Failed to read from socket")]
-    ReadError(#[from] io::Error),
-    /// Error when decoding the raw buffer into a protobuf.
-    #[error("Failed to decode protobuf")]
-    ProtobufError(#[from] DecodeError),
+pub enum FloodsubError {
+    /// Error from prost_codec::Codec
+    #[error(transparent)]
+    Codec(#[from] prost_codec::Error),
     /// Error when parsing the `PeerId` in the message.
     #[error("Failed to decode PeerId from message")]
     InvalidPeerId,
+    /// Error when decoding the raw buffer into a protobuf.
+    #[error("Failed to decode protobuf")]
+    ProtobufError(#[from] EncodeDecodeError),
+    /// Error when reading the packet from the socket.
+    #[error("Failed to read from socket")]
+    ReadError(#[from] io::Error),
 }
 
 #[derive(thiserror::Error, Debug)]
 #[error(transparent)]
-pub struct DecodeError(prost::DecodeError);
+pub struct EncodeDecodeError(prost_codec::Error);
```

Comment:
```suggestion
pub struct CodecError(prost_codec::Error);
```

Reply:
haha! I put that name down first, but changed my mind about 5 times before settling. I agree and will change it.